flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/DataStream.java

```diff
@@ -1303,6 +1305,22 @@ public ExecutionConfig getExecutionConfig() {
 		return sink;
 	}
 
+	/**
+	 * Adds the given {@link Sink} to this DataStream. Only streams with sinks added
+	 * will be executed once the {@link StreamExecutionEnvironment#execute()} method is called.
+	 *
+	 * @param sink The user defined sink.
+	 *
+	 * @return The closed DataStream.
+	 */
+	@Experimental
+	public DataStreamSink<T> addSink(Sink<T, ?, ?, ?> sink) {
```

Comment:
Why not renaming it to `toSink()`? this is aligned with the new Sources and also it will make the scala compiler happy. If not, we are breaking users' code.